firmware/controllers/algo/dynoview.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * @file dynoview.cpp
+ *
+ * @date Nov 29, 2020
+ * @author Alexandru Miculescu, (c) 2012-2020
+ */
+
+#include "engine.h"
+
+#if EFI_DYNO_VIEW
+#include "dynoview.h"
+#include "vehicle_speed.h"
+
+static Logging *logger;
+
+#if EFI_TUNER_STUDIO
+#include "tunerstudio_outputs.h"
+extern TunerStudioOutputChannels tsOutputChannels;
+#endif /* EFI_TUNER_STUDIO */
+
+EXTERN_ENGINE;
+
+DynoViewBase dynoInstance;
+
+void DynoViewBase::update(vssSrc src){
+
+    efitimeus_t timeNow, deltaTime = 0.0;
+    float speed,deltaSpeed = 0.0;
+    timeNow = getTimeNowUs();
+    speed = getVehicleSpeed();
+    if (src == ICU) {
+        speed = efiRound(speed,1.0);
+    } else {
+        //use speed with 0.1 precision
+        speed = efiRound(speed,0.001);
+    }
+
+    if(timeStamp != 0) {
+
+        if ( vss!= speed ) {
```

Comment:
i have to say that your code formatting is all over the place in terms of spaces

" if(timeStamp != 0) {" vs "if ( vss!= speed ) {" vs "if (vss > speed) {"

please note that this is very noticeable